pkg/github/token.go

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gofri/go-github-ratelimit/github_ratelimit"
+	"github.com/google/go-github/v41/github"
+	"golang.org/x/oauth2"
+)
+
+type GitHubToken interface {
+	GetNewGitHubClient() (*github.Client, error)
+}
+
+type GitHubTokenClient struct {
```

Comment:
Can we add a field here for the token name?  I need to be able to access it in the controllers when I add the metrics

Reply:
that can work too

Reply:
We can't add it to GitHubTokenClient, as it's just used for retrieving the token and generating the Go-GitHub clients

But I added a wrapper struct to the Go-GitHub client, which now includes the token name, along with the client itself.

Reply:
ok, that sounds good